snuba/web/ast_rollout.py

```diff
@@ -0,0 +1,47 @@
+from typing import Optional
+import random
+
+from snuba import settings
+from snuba.state import get_config
+
+ROLLOUT_RATE_CONFIG = "ast_rollout_rate"
+KILLSWITCH_CONFIG = "ast_shutdown"
+
+
+def is_ast_rolled_out(dataset_name: str, referrer: Optional[str]) -> bool:
+    """
+    Applies some rules to decide whether we should run a query based
+    on the AST or on the legacy representation.
+
+    - if the emergency killswitch is on, then no rollout
+    - if a dataset is rolled out then use ast
+    - if the referrer is rolled out for a dataset then use ast
+    - if none of the above decide according to the master
+        rollout percentage
+    """
+
+    if get_config(KILLSWITCH_CONFIG, 0):
```

Comment:
Will this be set to 0 on deployment?

Reply:
No. This is the emergency killswitch if something goes wrong during weekend or at night while we already started rolling out and when doing a deployment is impractical. 
We should never touch this ideally.